docs/package.json

```diff
@@ -5,6 +5,97 @@
   "author": "Material-UI Team",
   "license": "MIT",
   "scripts": {
-    "typescript": "tslint -p tsconfig.json \"src/pages/**/*.{ts,tsx}\""
+    "build": "rimraf .next && cross-env NODE_ENV=production next build",
+    "build-sw": "node ./scripts/buildServiceWorker.js",
+    "dev": "rimraf ../node_modules/.cache/babel-loader && cross-env NODE_ENV=development node src/server.js",
+    "deploy": "git push material-ui-docs master:latest",
+    "export": "rimraf docs/export && next export -o export && yarn build-sw && cp -r netlify/. export",
+    "icons": "rimraf static/icons/* && node ./scripts/buildIcons.js",
+    "start": "next start",
+    "typescript": "tslint -p tsconfig.json \"src/pages/**/*.{ts,tsx}\"",
+    "typescript:transpile": "node scripts/formattedTSDemos",
+    "typescript:transpile:dev": "node scripts/formattedTSDemos --watch"
+  },
+  "dependencies": {
```

Comment:
Do we need to make a distinction between the dev dependencies and dependencies?

Reply:
We never "need" to do this. It's just helpful to filter what dependencies are used in production and what are used for auxiliary tasks.

`dependencies` is used just as you would in your normal `nextjs` app. `devDependencies` are used in scripts.

Remember one of the goals of this PR: Make focused installs as small as possible.

Reply:
In the future, how do I know where a new dependency should be added? For instance, are the `doctrine`, `typescript-to-proptypes`, `address` and `@types` dependencies necessary to run the documentations in production?

> Make focused installs as small as possible.

If I understand correctly, we can only optimize for one case. Do you want to optimize for reducing the time it takes to run the tests in the CI or the time it takes to deploy the documentation in Netlify?

Reply:
> If I understand correctly, we can only optimize for one case. 

This sounds like a tradeoff which it isn't. It is strictly better to differentiate between devDependencies and dependencies. 

>  Do you want to optimize for reducing the time it takes to run the tests in the CI or the time it takes to deploy the documentation in Netlify

Reduce azure builds and netlify deploys. CircleCI times should not be affected by this change.

> In the future, how do I know where a new dependency should be added?

Does `next build` pass without this dependency? => `devDependency`. When in doubt: put into dependencies.

`@types` are there because of `next@9`. Will see if we can move them to dev in the future.

Reply:
Ok sounds good, I'm curious to see the impact.

Reply:
Actual features are likely only available in yarn v2. Only immediate benefit is moving docs "stuff" closer together.